packages/back-end/src/services/experiments.ts

```diff
@@ -669,6 +666,207 @@ export async function createSnapshot({
+    Array.from(analysisParamsMap.values())
+  );
+
+  results.map(async (result) => {
+    const analysisParams = analysisParamsMap.get(result.id);
+    if (!analysisParams) return;
+
+    const {
+      analysis,
+      analyses,
+      snapshotSettings,
+      queryResults,
+      organization,
+      snapshot,
+    } = analysisParams;
+    let { unknownVariations } = analysisParams;
+
+    // TODO fix for dimension slices and move to health query
+    const multipleExposures = Math.max(
+      ...queryResults.map(
+        (q) =>
+          q.rows.filter((r) => r.variation === "__multiple__")?.[0]?.users || 0
+      )
+    );
+
+    const experimentReportResults: ExperimentReportResults[] = [];
+    analyses.forEach((_, i) => {
+      const dimensionMap: Map<
+        string,
+        ExperimentReportResultDimension
+      > = new Map();
+      // TODO check for error
+      result.results.forEach(({ metric, analyses }) => {
+        // each result can have multiple analyses (a set of computations that
+        // use the same snapshot)
+        // we loop over the analyses requested and pull out the results for each one
+        const result = analyses[i];
+        if (!result) return;
+
+        unknownVariations = unknownVariations.concat(result.unknownVariations);
+
+        result.dimensions.forEach((row) => {
+          const dim = dimensionMap.get(row.dimension) || {
+            name: row.dimension,
+            srm: 1,
+            variations: [],
+          };
+
+          row.variations.forEach((v, i) => {
+            const data = dim.variations[i] || {
+              users: v.users,
+              metrics: {},
+            };
+            data.users = Math.max(data.users, v.users);
+            data.metrics[metric] = {
+              ...v,
+              buckets: [],
+            };
+            dim.variations[i] = data;
+          });
+
+          dimensionMap.set(row.dimension, dim);
+        });
+      });
+
+      const dimensions = Array.from(dimensionMap.values());
+      if (!dimensions.length) {
+        dimensions.push({
+          name: "All",
+          srm: 1,
+          variations: [],
+        });
+      } else {
+        dimensions.forEach((dimension) => {
+          // Calculate SRM
+          dimension.srm = checkSrm(
+            dimension.variations.map((v) => v.users),
+            snapshotSettings.variations.map((v) => v.weight)
+          );
+        });
+      }
+      experimentReportResults.push({
+        multipleExposures,
+        unknownVariations: Array.from(new Set(unknownVariations)),
+        dimensions,
+      });
+    });
+
+    if (result.error) {
+      analysis.results = [];
+      analysis.status = "error";
+      analysis.error = result.error;
+    } else {
+      analysis.results = experimentReportResults[0]?.dimensions || [];
+      analysis.status = "success";
+      analysis.error = undefined;
+    }
+
+    updateSnapshotAnalysis({ organization, id: snapshot, analysis, context });
```

Comment:
This is running in the background and not being awaited.  If that's intentional, then add a `.catch` here so we won't have errors for unhandled promise rejections.  Otherwise, if you do want to await, then need to queue up the promises in an array and use `promiseAllChunks` to batch them to avoid sending too many updates to Mongo at once.

Reply:
Changed to an array of promises and chunked into 10 at a time.